### PR TITLE
python312Packages.torchmetrics: 1.4.3 -> 1.6.1

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Things done

Changelogs:
- https://github.com/Lightning-AI/torchmetrics/releases/tag/v1.5.0
- https://github.com/Lightning-AI/torchmetrics/releases/tag/v1.5.1
- https://github.com/Lightning-AI/torchmetrics/releases/tag/v1.6.0
- https://github.com/Lightning-AI/torchmetrics/releases/tag/v1.6.1

cc @SomeoneSerge 

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 350488`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 6 packages marked as broken and skipped:</summary>
  <ul>
    <li>piper-train</li>
    <li>piper-train.dist</li>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 18 packages built:</summary>
  <ul>
    <li>python311Packages.finetuning-scheduler</li>
    <li>python311Packages.finetuning-scheduler.dist</li>
    <li>python311Packages.pyannote-audio</li>
    <li>python311Packages.pyannote-audio.dist</li>
    <li>python311Packages.pytorch-lightning</li>
    <li>python311Packages.pytorch-lightning.dist</li>
    <li>python311Packages.torchmetrics</li>
    <li>python311Packages.torchmetrics.dist</li>
    <li>python312Packages.finetuning-scheduler</li>
    <li>python312Packages.finetuning-scheduler.dist</li>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>python312Packages.pytorch-lightning</li>
    <li>python312Packages.pytorch-lightning.dist</li>
    <li>python312Packages.torchmetrics</li>
    <li>python312Packages.torchmetrics.dist</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 6 packages marked as broken and skipped:</summary>
  <ul>
    <li>piper-train</li>
    <li>piper-train.dist</li>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python311Packages.finetuning-scheduler</li>
    <li>python311Packages.finetuning-scheduler.dist</li>
    <li>python311Packages.pyannote-audio</li>
    <li>python311Packages.pyannote-audio.dist</li>
    <li>python311Packages.pytorch-lightning</li>
    <li>python311Packages.pytorch-lightning.dist</li>
    <li>python311Packages.torchmetrics</li>
    <li>python311Packages.torchmetrics.dist</li>
    <li>python312Packages.finetuning-scheduler</li>
    <li>python312Packages.finetuning-scheduler.dist</li>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>python312Packages.pytorch-lightning</li>
    <li>python312Packages.pytorch-lightning.dist</li>
    <li>python312Packages.torchmetrics</li>
    <li>python312Packages.torchmetrics.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.finetuning-scheduler</li>
    <li>python311Packages.finetuning-scheduler.dist</li>
    <li>python312Packages.finetuning-scheduler</li>
    <li>python312Packages.finetuning-scheduler.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.pytorch-lightning</li>
    <li>python311Packages.pytorch-lightning.dist</li>
    <li>python311Packages.torchmetrics</li>
    <li>python311Packages.torchmetrics.dist</li>
    <li>python312Packages.pytorch-lightning</li>
    <li>python312Packages.pytorch-lightning.dist</li>
    <li>python312Packages.torchmetrics</li>
    <li>python312Packages.torchmetrics.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 8 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.finetuning-scheduler</li>
    <li>python311Packages.finetuning-scheduler.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
    <li>python312Packages.finetuning-scheduler</li>
    <li>python312Packages.finetuning-scheduler.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.pyannote-audio</li>
    <li>python311Packages.pyannote-audio.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.pytorch-lightning</li>
    <li>python311Packages.pytorch-lightning.dist</li>
    <li>python311Packages.torchmetrics</li>
    <li>python311Packages.torchmetrics.dist</li>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>python312Packages.pytorch-lightning</li>
    <li>python312Packages.pytorch-lightning.dist</li>
    <li>python312Packages.torchmetrics</li>
    <li>python312Packages.torchmetrics.dist</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>

**Comment by SomeoneSerge** [comment]
aarch64 failure is new https://hydra.nixos.org/build/276257303

**Comment by GaetanLepage** [comment]
> aarch64 failure is new https://hydra.nixos.org/build/276257303

Indeed :'(
I have no idea why though:
```
terminate called after throwing an instance of 'onnxruntime::OnnxRuntimeException'
  what():  /build/source/include/onnxruntime/core/common/logging/logging.h:320 static const onnxruntime::logging::Logger& onnxruntime::logging::LoggingManager::DefaultLogger() Attempt to use DefaultLogger but none has been registered.
```

**Comment by GaetanLepage** [comment]
I bisected it to [this commit](https://github.com/Lightning-AI/torchmetrics/commit/1ac0b63677dd720dd453772239bbc63fe1562d6a).

What is very strange is that this bug happens only:
- on `aarch64-linux`
- within a 3-layer deep downstream dependency (`torchmetrics` -> `pytorch-lightning` -> `pyannote-audio` -> `whisper-ctranslate2`).

**Comment by autrimpo** [comment]
FYI it builds fine for me with sandbox disabled.

**Review by SomeoneSerge — APPROVED** [review]
Didn't check the runtime, but the tests pass, the diff looks good, not sure why the draft status
